source/common/event/dispatcher_impl.cc

```diff
@@ -169,15 +169,16 @@ Network::DnsResolverSharedPtr DispatcherImpl::createDnsResolver(
         "Apple's API only allows overriding DNS resolvers via system settings. Delete resolvers "
         "config or disable the envoy.restart_features.use_apple_api_for_dns_lookups runtime "
         "feature.");
-    RELEASE_ASSERT(!use_tcp_for_dns_lookups,
+    RELEASE_ASSERT(!dns_resolver_options_.has_use_tcp_for_dns_lookups() ||
+                       !dns_resolver_options.use_tcp_for_dns_lookups().value(),
```

Comment:
Be sure to run "check_format fix"

Reply:
yes, the check_format fix is taken care.👍

removed "has_use_tcp_for_dns_lookups" which is redundant here as well.